### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Chimoneg27**
### For this project I did the following

- Use ruby gems as software packages system.
- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Use params from browser request in a safe way.
- Use preprocessed html file with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a webapp that requires the user to log in.
- Use devise gem for authentication.
- Limit access to webapp resources based on authorization rules.
- Analyze in writing why you have made a coding choice using one structure over another.
- Added live demo link and link to my video walkthrough

### **Notice**

**My tests run well, check the video below. They do take a while, but they run. **

https://github.com/Chimoneg27/Rails-Budget-App/assets/106396944/54fe6a65-225f-4a10-ba85-8233193a7824

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,10 @@ Key features of the application.
 - A user can create a purchase for a group
 - User authentication
 - Splash page
+- User can view all purchases
+- User can view all groups
```
Comment (omar-muhamad) [inline README.md:58]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#design:~:text=the%20author.%20Therefore%2C-,you%20must%20do%20it%20in%20the%20README%20of%20your%20project.,-Interactions), Pleese give appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) at the `Acknowledgments` section in your README file.

- [ ] The MIT license link in the readme file is broken, please provide a valid link.

image

File: README.md [hunk 2]
```diff
@@ -54,6 +54,10 @@ Key features of the application.
...
+- User can view all purchases for a group
+- Login and signup links on splash page
```
Comment (omar-muhamad) [inline README.md:61]: As per the project [description](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=about%20building%20a-,mobile%20web%20application,-where%20you%20can) and the given design guidelines, This app should be a mobile app with mobile view, so please consider the following changes:

- [ ] At the categories page(Home page), Please add the correct placement for all page elements (Layout: composition and space between elements) and also add the correct font family, font sizes, and font weight for all elements too to fulfill the project required [design guidelines](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#design).

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

- [ ] Same here at the transaction page, Please add the correct placement for all page elements (Layout: composition and space between elements) and also add the correct font family, font sizes, and font weight for all elements too to fulfill the project required [design guidelines](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#design).

| Given Design  | Your Design |
| ------------- | ------------- |
| image  | image  |

- [ ] When I run the specs you add, It takes forever to run and never return anything (success or failed), Please fix that.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Chimoneg27,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Chimoneg27** [comment]
@Omar-Muhamad did you view the app on your phone. Is it supposed to be responsive

**Comment by Chimoneg27** [comment]
@Omar-Muhamad the specs taking time to run isn't supposed to be an issue. We were supposed to write tests. Not to improve their speed. This is not what we were taught.

**Comment by omar-muhamad** [comment]
Hi @Chimoneg27,
I hope that you're fine.

Yes I tested your app carefully and the design of the mobile view is far from the given design guidelines in both layout and typography (see the included screenshots), so please consider making the required changes I mentioned above.

For the specs, Yes you should make specs that run seamlessly and successfully. I tested  all specs at once and every one individually and it seems that there are test cases that make the compiler go in an infinity loop and never return a result, so Kindly fix that.

Have a great day.

**Review by mckent05 — APPROVED** [review]
Hi @Chimoneg27, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job working on your capstone project 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**